### PR TITLE
feat: add all http methods

### PR DESCRIPTION
**Pull request by HeadTriXz**

### CODE REVIEW

**Review by tbnritzdoge — APPROVED** [review]
lgtm

File: src/lib/PetitioRequest.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ import { stringify } from "querystring"; // eslint-disable-line no-duplicate-imp
 /**
  * Accepted HTTP methods (currently only supports up to HTTP/1.1).
  */
-export type HTTPMethod = "GET" | "POST" | "PATCH" | "DELETE" | "PUT";
+export type HTTPMethod = "GET" | "HEAD" | "POST" | "PATCH" | "DELETE" | "PUT";
```
Comment (lveneris) [inline src/lib/PetitioRequest.ts]:
```suggestion
export type HTTPMethod = "GET" | "HEAD" | "POST" | "OPTIONS" | "PUT" | "DELETE" | "TRACE" | "CONNECT" | "PATCH";
```

**Review by Nytelife26 — CHANGES_REQUESTED** [review]
Many thanks for the PR! I think while we're at it, we might as well add all the other request methods from HTTP/1.0 and HTTP/1.1 too, so if you can commit these we're good to go :)

**Review by tbnritzdoge — APPROVED** [review]

**Review by Vetlix — APPROVED** [review]
LGTM

**Review by Nytelife26 — APPROVED** [review]
Thank you. Merging now :)

**Pull request merged**